velox/docs/develop/aggregate-functions.rst

```diff
@@ -169,6 +184,14 @@ function's argument type(s) wrapped in a Row<> even if the function only takes
 one argument. This is needed for the SimpleAggregateAdapter to parse input
 types for arbitrary aggregation functions properly.
 
+Some function-level variables needs to be declared in the simple aggregation
+function class. These variables are initialized once when the aggregation
+function is created and used at every row when adding inputs to accumulators
+or extracting values from accumulators. For example, if the aggregation
+function needs to get the result type or the raw input type of the aggregation
+function, the author can hold them in the aggregate class variables, and
+initialize them in the initialize() method.
```

Comment:
nit: the author can hold them in the aggregate class variables, and
initialize them in the initialize() method. --> these types can be defined as member variables in the aggregate class and initialized in the initialize() method.